hack/local-up-cluster.sh

```diff
@@ -1044,6 +1044,93 @@ if [[ "${KUBETEST_IN_DOCKER:-}" == "true" ]]; then
 
   # kubekins has a special directory for docker root
   DOCKER_ROOT="/docker-graph"
+  # enable cri 
+  # restart docker
+  # enable ipv6
+  # install cni config
+  # install cni 
+  # restart docker
+
+  echo "enable cri"
+  echo "DOCKER_OPTS=\"\${DOCKER_OPTS} --cri-containerd\"" >> /etc/default/docker
+
+  echo "restarting docker"
+  service docker restart 
+
+  echo "install cni"
+
+  CNI_DIR=${DESTDIR:=''}/opt/cni
+  CNI_CONFIG_DIR=${DESTDIR}/etc/cni/net.d
+
+  PDIR=$(pwd)
+  cd /workspace 
+  TMPROOT=$(mktemp -d)
+  git clone https://github.com/containernetworking/plugins.git "${TMPROOT}"/plugins
+  pushd "${TMPROOT}"/plugins
+  git checkout v1.0.1
+  ./build_linux.sh
+  mkdir -p $CNI_DIR
+  cp -r ./bin $CNI_DIR
+  mkdir -p $CNI_CONFIG_DIR
+  cat << EOF | tee $CNI_CONFIG_DIR/10-containerd-net.conflist
+{
+  "cniVersion": "1.0.0",
+  "name": "containerd-net",
+  "plugins": [
+    {
+      "type": "bridge",
+      "bridge": "cni0",
+      "isGateway": true,
+      "ipMasq": true,
+      "promiscMode": true,
+      "ipam": {
+        "type": "host-local",
+        "ranges": [
+          [{
+            "subnet": "10.88.0.0/16"
+          }],
+          [{
+            "subnet": "2001:4860:4860::/64"
+          }]
+        ],
+        "routes": [
+          { "dst": "0.0.0.0/0" },
+          { "dst": "::/0" }
+        ]
+      }
+    },
+    {
+      "type": "portmap",
+      "capabilities": {"portMappings": true}
+    }
+  ]
+}
+EOF
+popd
+rm -fR "${TMPROOT}"
+
+cd $PDIR
+echo "Enabling IPV6 for Docker."
+# configure the daemon with ipv6
+mkdir -p /etc/docker/
+    cat <<EOF >/etc/docker/daemon.json
+{
+  "ipv6": true,
+  "fixed-cidr-v6": "fc00:db8:1::/64"
+}
+EOF
```

Comment:
why is docker required? is not about containerd only?

Reply:
we need docker to run kubetest